@here/harp-mapview/lib/MapView.ts

```diff
@@ -2956,6 +2964,40 @@ export class MapView extends THREE.EventDispatcher {
         const worldOffsetX = tile.computeWorldOffsetX();
         if (tile.willRender(zoomLevel)) {
             for (const object of tile.objects) {
+                if (object.userData.technique && object.userData.feature) {
+                    const technique: Technique = object.userData.technique;
+                    const objInfos = object.userData.feature.objInfos as ValueMap[];
+                    if (technique.hasOwnProperty("filter") && Array.isArray(objInfos)) {
+                        // ### TODO: evaluate only when properties references by the filter change.
+                        const geometry: THREE.BufferGeometry = (object as any).geometry;
+                        if (geometry && geometry.isBufferGeometry) {
+                            geometry.clearGroups();
+                            objInfos.forEach((properties, index) => {
+                                // ### TODO: evaluate only when the runtime dependencies
+                                // of the incoming feature are different from the one
+                                // used by the previous evaluation.
+                                const env = new MapEnv(properties, this.m_env);
+                                const result = getPropertyValue(technique.filter, env) ?? false;
+                                if (result === true) {
+                                    const start = object.userData.feature.starts[index] as number;
+                                    const end =
+                                        object.userData.feature.starts[index + 1] ??
+                                        geometry.getIndex()?.count;
+                                    if (end === undefined) {
+                                        return;
+                                    }
+                                    const count = end - start;
+                                    geometry.addGroup(start, count);
+                                }
```

Comment:
the dynamic filtering/creation of `geometry groups` is assuming that the `geometries` created to represent `geo features` have shared buffer attributes.